pol-core/pol/realms/realmfunc.cpp

```diff
@@ -624,73 +614,75 @@ bool Realm::dropheight( Plib::MapShapeList& shapes, short dropz, short chrz, sho
                                                       // the house is being edited,
           // everyone in the house would use it for walking...
-          multi->readshapes( vec, s16( x ) - multi->x(), s16( y ) - multi->y(), multi->z() );
+          multi->readshapes( vec, delta.x(), delta.y(), multi->z() );
         else
         {
           const Multi::MultiDef& def = multi->multidef();
-          def.readshapes( vec, s16( x ) - multi->x(), s16( y ) - multi->y(), multi->z(), anyflags );
+          def.readshapes( vec, delta.x(), delta.y(), multi->z(), anyflags );
         }
       } );
 }
 
-void Realm::readmultis( Plib::MapShapeList& vec, unsigned short x, unsigned short y,
-                        unsigned int anyflags, MultiList& mvec ) const
+void Realm::readmultis( Plib::MapShapeList& vec, const Core::Pos2d& pos, unsigned int anyflags,
+                        MultiList& mvec ) const
 {
   Core::WorldIterator<Core::MultiFilter>::InRange(
-      x, y, this, 64,
+      pos, this, 64,
       [&]( Multi::UMulti* multi )
       {
         Multi::UHouse* house = multi->as_house();
+        Core::Vec2d delta = pos - multi->pos().xy();
         if ( house != nullptr && house->IsCustom() )
         {
-          if ( multi->readshapes( vec, s16( x ) - multi->x(), s16( y ) - multi->y(), multi->z() ) )
+          if ( multi->readshapes( vec, delta.x(), delta.y(), multi->z() ) )
             mvec.push_back( multi );
         }
         else
         {
           const Multi::MultiDef& def = multi->multidef();
-          if ( def.readshapes( vec, s16( x ) - multi->x(), s16( y ) - multi->y(), multi->z(),
-                               anyflags ) )
+          if ( def.readshapes( vec, delta.x(), delta.y(), multi->z(), anyflags ) )
           {
             mvec.push_back( multi );
           }
         }
       } );
 }
 
-void Realm::readmultis( Plib::StaticList& vec, unsigned short x, unsigned short y ) const
+void Realm::readmultis( Plib::StaticList& vec, const Core::Pos2d& pos ) const
 {
   Core::WorldIterator<Core::MultiFilter>::InRange(
-      x, y, this, 64,
+      pos, this, 64,
       [&]( Multi::UMulti* multi )
       {
         Multi::UHouse* house = multi->as_house();
+        Core::Vec2d delta = pos - multi->pos().xy();
         if ( house != nullptr && house->IsCustom() )  // readshapes switches to working design if
                                                       // the house is being edited,
           // everyone in the house would use it for walking...
-          multi->readobjects( vec, int( x ) - multi->x(), int( y ) - multi->y(), multi->z() );
+          multi->readobjects( vec, delta.x(), delta.y(), multi->z() );
         else
         {
           const Multi::MultiDef& def = multi->multidef();
-          def.readobjects( vec, int( x ) - multi->x(), int( y ) - multi->y(), multi->z() );
+          def.readobjects( vec, delta.x(), delta.y(), multi->z() );
```

Comment:
Just as a note to the future: this code smells. I don't know why we need to handle custom houses separately from other multis.